core-ui/src/components/Predefined/Create/Secrets/CreateSecretForm.js

```diff
@@ -1,97 +1,223 @@
-import React, { useState } from 'react';
-import LuigiClient from '@luigi-project/client';
+import React, { useState, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
+import * as jp from 'jsonpath';
+import {
+  Button,
+  FormInput,
+  FormTextarea,
+  ComboboxInput,
+} from 'fundamental-react';
+import { Tooltip } from 'react-shared';
 
+import { useMicrofrontendContext } from 'react-shared';
+import { base64Decode, base64Encode } from 'shared/helpers';
+import { ResourceForm } from 'shared/ResourceForm/ResourceForm';
 import {
-  usePost,
-  useNotification,
-  useMicrofrontendContext,
-} from 'react-shared';
-import { CreateForm } from 'shared/components/CreateForm/CreateForm';
+  MultiInput,
+  K8sNameField,
+  KeyValueField,
+} from 'shared/ResourceForm/components/FormComponents';
+
 import {
-  secretToYaml,
-  yamlToSecret,
   createSecretTemplate,
   createPresets,
+  readFromFile,
+  getSecretDefs,
 } from './helpers';
-import { SimpleForm } from './SimpleForm';
-import { AdvancedForm } from './AdvancedForm';
+
+import './CreateSecretForm.scss';
 
 export function CreateSecretForm({
   namespaceId,
   formElementRef,
```

Comment:
`formElementRef`, `onChange` and `onSubmit` are not used anywhere...and probably should be used in `ResourceForm`